scripts/src/javatests/org/oppia/android/scripts/maven/GenerateMavenDependenciesListTest.kt

```diff
@@ -0,0 +1,664 @@
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  private val mockLicenseFetcher by lazy { initializeLicenseFetcher() }
+  private val commandExecutor by lazy { initiazeCommandExecutorWithLongProcessWaitTime() }
+  private lateinit var testBazelWorkspace: TestBazelWorkspace
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("scripts", "assets")
+    tempFolder.newFolder("third_party")
+    testBazelWorkspace = TestBazelWorkspace(tempFolder)
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testEmptyPbFile_scriptFailsWithException_writesTextproto() {
+    val textprotoFile = tempFolder.newFile("scripts/assets/maven_dependencies.textproto")
+    tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+
+    val coordsList = listOf(DEP_WITH_SCRAPABLE_LICENSE, DEP_WITH_DIRECT_LINK_ONLY_LICENSE)
+    setupBazelEnvironment(coordsList)
+
+    val exception = assertThrows(Exception::class) {
+      GenerateMavenDependenciesList(
+        mockLicenseFetcher,
+        commandExecutor
+      ).main(
+        arrayOf(
+          "${tempFolder.root}",
+          "scripts/assets/maven_install.json",
+          "scripts/assets/maven_dependencies.textproto",
+          "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+        )
+      )
+    }
+    assertThat(exception).hasMessageThat().contains(LICENSE_DETAILS_INCOMPLETE_FAILURE)
+
+    val mavenDependencyList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addAllLicense(
+          listOf(
+            License.newBuilder().apply {
+              this.licenseName = "The Apache License, Version 2.0"
+              this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+            }.build()
+          )
+        )
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_DIRECT_LINK_ONLY_LICENSE
+        this.artifactVersion = FIREBASE_ANALYTICS_VERSION
+        this.addAllLicense(
+          listOf(
+            License.newBuilder().apply {
+              this.licenseName = "Android Software Development Kit License"
+              this.originalLink = "https://developer.android.com/studio/terms.html"
+            }.build()
+          )
+        )
+      }.build()
+    )
+    val expectedMavenDependencyList = MavenDependencyList.newBuilder().apply {
+      this.addAllMavenDependency(mavenDependencyList)
+    }.build()
+    val outputMavenDependencyList = getProto(
+      textprotoFile,
+      MavenDependencyList.getDefaultInstance()
+    )
+    assertThat(outputMavenDependencyList).isEqualTo(expectedMavenDependencyList)
+  }
```

Comment:
@BenHenning confirming if this looks good.

Reply:
Ah I see. I actually think I lean toward the individual property checking with a helper since it's a lot more explicit on the defaults. Here, we're hiding default properties vs. being explicit about what we intend them to be.

Both are technically equivalent, but it's good to be clearer in tests.